subprojects/core/src/main/java/org/gradle/internal/scripts/BuildScriptCompileAndInstrumentUnitOfWork.java

```diff
@@ -34,36 +38,40 @@
 import static java.util.Objects.requireNonNull;
 
 /**
- * A base class that represents a work for compilation for Kotlin and Groovy scripts.
+ * A base class that represents a work for compilation for Kotlin and Groovy build scripts.
+ * This work unit first compiles the build script to a directory, and then instruments the directory for configuration cache and returns instrumented output.
  */
-public abstract class BuildScriptCompileUnitOfWork implements ImmutableUnitOfWork {
+public abstract class BuildScriptCompileAndInstrumentUnitOfWork implements ImmutableUnitOfWork {
```

Comment:
❓ How about `BuildScriptCompilationAndInstrumentation`? We don't have the `UnitOfWork` suffix in `TaskExecution` nor in `TransformExecution`. And we use the noun form, not verbs, so "execution" instead of "execute".

Well, it's not entirely universal, but I think this is the pattern we want to follow, and change the rest of the names accordingly.

Reply:
Sounds good, renamed